src/Maestro/Maestro.Web/MaestroServiceHost.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.ApplicationInsights.Channel;
+using System.Fabric.Health;
+using System.Fabric;
+using System.Net;
+using System.Threading;
+using System;
+using Microsoft.DotNet.ServiceFabric.ServiceHost;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Hosting;
+
+namespace Maestro.Web;
+
+public class MaestroServiceHost : ServiceHost
+{
+    private MaestroServiceHost() { }
+
+    /// <summary>
+    ///     Configure and run a new ServiceHost
+    /// </summary>
+    public new static void Run(Action<ServiceHost> configure)
```

Comment:
This method is almost the same as the one we override. Cannot we just abstract away the part where it differs into a protected method/property that we only override?

What are the parts where this differs?

As an example why this approach is not great because in 1 month when we are moving to TLS 1.3, someone will change this in the shared library but then we need to do the same here somehow. It's quite a lot of infra code that we need to maintain in two places.

Reply:
the only difference is the `host = new ...`, originally it's creating a ServiceHost, but we need it to initialize the new class, so the new method can be called